server/service/src/validate.rs

```diff
@@ -64,6 +65,12 @@ pub fn check_other_party(
                 return Err(OtherPartyErrors::TypeMismatched);
             }
         }
+
+        CheckOtherPartyType::Patient => {
+            if other_party.name_row.r#type != NameType::Patient {
```

Comment:
Could probably add an other_party.is_patient() function like above, but would require OG changes.

Reply:
what do you mean with OG changes, can't you just put this check into a `is_patient()` method?

Reply:
[05f0340](https://github.com/openmsupply/open-msupply/pull/1965/commits/05f03401d53eba2daaf9c44db66dbb5f033ea4a1)

I was thinking of having it in the name_store_join 🤔 , but doesn't seem to correctly map to null correctly from OG, so have added this instead (: